### PR TITLE
Add read only validation to read only fields

### PR DESCRIPTION
**Pull request by ahidalgob**
Add read only validation to DagRunEditForm and TaskInstanceEditForm read only fields.

### CODE REVIEW

**Comment by ahidalgob** [comment]
It would be nice to have a base airflow form Class so we could modify `Form.populate_obj` there and not for every form that has a `ReadOnly` field, but I'm not sure the change it's worth it given only 2 forms use this.

**Comment by hussein-awala** [comment]
Actually it is not related to this change, and it was fixed by #33416, I merge main

File: airflow/www/forms.py [hunk 1]
```diff
@@ -124,21 +124,28 @@ class DagRunEditForm(DynamicForm):
...
+    run_id = StringField(lazy_gettext("Run Id"), validators=[ReadOnly()], widget=BS3TextFieldROWidget())
+    state = StringField(lazy_gettext("State"), validators=[ReadOnly()], widget=BS3TextFieldROWidget())
     execution_date = DateTimeWithTimezoneField(
         lazy_gettext("Logical Date"),
+        validators=[ReadOnly()],
         widget=AirflowDateTimePickerROWidget(),
     )
-    conf = TextAreaField(lazy_gettext("Conf"), widget=BS3TextAreaROWidget())
+    conf = TextAreaField(lazy_gettext("Conf"), validators=[ReadOnly()], widget=BS3TextAreaROWidget())
     note = TextAreaField(lazy_gettext("User Note"), widget=BS3TextAreaFieldWidget())
```
Comment (uranusjr) [inline airflow/www/forms.py:142]: This is editable right? So the class docstring should be modified a bit?

**Review by potiuk — APPROVED** [review]
Nice way of implementing th read-only behaviour :)

**Pull request merged**